src/Core/Logger.hpp

```diff
@@ -0,0 +1,40 @@
+//
+//  ScriptingManager.hpp
+//  embedding_test
+//
+//  Created by Leonidas Neftali Gonzalez Campos on 25/12/23.
+//
+
+#pragma once
+
+#define _SET_TEXT_COLOR(COLOR, STREAM) fprintf(STREAM, "\033[" COLOR "m")
+#define _RESET_COLOR(STREAM) fprintf(STREAM, "\033[0m")
+
+#define _LOG_LN(FORMAT, ...) fprintf(stdout, FORMAT, ##__VA_ARGS__); \
+                        fprintf(stdout, "\n\r")
+
+#define _LOG_ERROR_LN(FORMAT, ...) fprintf(stderr, FORMAT, ##__VA_ARGS__); \
+    							fprintf(stderr, "\n\r")
+
+#define LOG_INFO_LN(FORMAT, ...) _SET_TEXT_COLOR("1;32", stdout); \
+							fprintf(stdout, "[Info] - "); \
+                            _LOG_LN(FORMAT, ##__VA_ARGS__); \
+							_RESET_COLOR(stdout)
+
+#define LOG_WARN_LN(FORMAT, ...) _SET_TEXT_COLOR("1;33", stdout); \
+							fprintf(stdout, "[Warn] - "); \
+                            _LOG_LN(FORMAT, ##__VA_ARGS__); \
+							_RESET_COLOR(stdout)
+
+#if DEBUG
+#define LOG_DEBUG_LN(FORMAT, ...) _SET_TEXT_COLOR("1;34", stdout); \
+							fprintf(stdout, "[Debug] - "); \
```

Comment:
I think that we should make a better log system, especially since some platforms don't allow us to watch stderr or stdout (such as Nintendo Switch), instead, we should abstract the destination into a sink, a composable component that handles the writing aspect of logging to something (a memory buffer, a file, a network socket, etc.)

Reply:
Wrote this to have a simple logger for now, let's discuss another implementation approach to abstract it the way you mention and avoid platform compatibility issues